### ISSUE TITLE
attach_view (maybe some weird issue)

### ISSUE THREAD

**Issue by SVC4230** [ValveSoftware/halflife#3248]
Hello!
A long time ago I experienced a problem with the native attach_view (PF_setview_I). That is why I wanted to report it and better understand why it happens.

Using the following code:
```pawn
#include    AMXMODX 
#include    ENGINE 

new bool:g_bInvertYaw; 

public plugin_init() 
{ 
    register_clcmd("__weapon__", "__give_weapon__"); 
    register_clcmd("invert_angle", "invert_angle"); 
} 

public invert_angle() g_bInvertYaw = !g_bInvertYaw; 

public __give_weapon__(const iPlayer) 
{ 
    // It's only for testing purpose; code can be improved a lot, I know :)... 
    
    new Float:fCoords[3]; 
    entity_get_vector(iPlayer, EV_VEC_origin, fCoords); 
    
    fCoords[2] += 35.0; 
    
    new iWeapon = create_entity("info_target"); 
    { 
        entity_set_origin(iWeapon, fCoords); 
        etity_set_model(iWeapon, "models/v_ak47.mdl"); 
        
        entity_set_string(iWeapon, EV_SZ_classname, "__custom_weapon__"); 
        entity_set_edict(iWeapon, EV_ENT_owner, iPlayer); 
        entity_set_float(iWeapon, EV_FL_nextthink, 0.001); 
        
        // I think doesn't necessary explain what it does.. 
        attach_view(iPlayer, iWeapon); 
        
        register_think("__custom_weapon__", "__think_custom_weapon__"); 
    } 
} 

public __think_custom_weapon__(const iEnt) 
{ 
    new iOwner = entity_get_edict(iEnt, EV_ENT_owner); 
    
    /* 
    Here I set custom-weapon's angles to the player's view-angles 
    However even if I reverse they (more precisely the yaw-angle), the up/down view will be reverse "again" 
    
    I can't explain better this, so when you test code you will see what i've saying 
    */ 
    
    static Float:fViewAngles[3]; 
    entity_get_vector(iOwner, EV_VEC_v_angle, fViewAngles); 
    
    // What I'm doing here is reverse the up/down angle (for testing) but doesn't not have effect 
    if (g_bInvertYaw) 
        fViewAngles[0] = -fViewAngles[0]; 
    
    entity_set_vector(iEnt, EV_VEC_angles, fViewAngles); 
    entity_set_float(iEnt, EV_FL_nextthink, 0.001); 
} 
```

If I look up, the "gun" will look down and vice versa.

Here are some images so you can see

Looking to the center
![Screenshot_20220222-103900_Chrome](https://user-images.githubusercontent.com/100200037/155167773-c99314e9-9bf8-42dc-a6d9-c9c0b07125b9.jpg)

Looking down
![Screenshot_20220222-103920_Chrome](https://user-images.githubusercontent.com/100200037/155167858-4fcb6f8f-bbe6-430e-9b42-95693fb1cc6b.jpg)

Looking up
![Screenshot_20220222-103910_Chrome](https://user-images.githubusercontent.com/100200037/155167915-1ae0b652-6124-4f7e-8989-4e9ef34edce5.jpg)

Basically the model is facing the wrong position when I raise/lower the camera. I do not know if the client's camera works that way when it is assigned to another entity different from the same.

It should be noted that it is only a code for testing, and currently I do not have a copy of the CS to test. However, I hope to receive any kind of response.
